ote/src/cljs/ote/views/main.cljs

```diff
@@ -55,31 +57,82 @@
                 :style style-base/flash-message
                 :auto-hide-duration 5000}])
 
+(defn- top-nav-links [e! app]
+
+  ;(.log js/console " when " (get app :width))
+  [:div
+  [:ul
+   [:li
+    [:a.main-icon {:href     "#"
+                   :on-click (fn [_] (set! (.-location js/document) "/"))}
+     [:img {:src "img/icons/nap-logo.svg"}]]]
+  [:li
+   [:a.ote-nav {:class    (is-topnav-active :front-page (:page app))
+                :href     "#"
+                :on-click (fn [_] (set! (.-location js/document) "/"))}
+    (tr [:common-texts :navigation-front-page])]]
+  [:li
+   [:a.ote-nav {:class    (is-topnav-active :front-page (:page app))
+                :href     "#"
+                :on-click (fn [_] (set! (.-location js/document) "/dataset"))}
+    "Palvelukatalogi"]]
+  [:li
+   [:a.ote-nav {:class    (is-topnav-active :front-page (:page app))
+                :href     "#"
+                :on-click (fn [_] (set! (.-location js/document) "/organization"))}
+    "Palveluntuottajat"]]
+  [:li
+   [:a.ote-nav {:class    (is-topnav-active :own-services (:page app))
+                :href     "#"
+                :on-click #(e! (fp-controller/->ChangePage :own-services))}
+    (tr [:common-texts :navigation-own-service-list])]]
+   ]
+  [:div.user-menu {:class (is-user-menu-active app)}
+   (r/as-element (user-menu e! (get-in app [:user :name])))]
+   ]
+  )
+
+(defn- mobile-top-nav-links [e! app]
+  [:ul
+    [:li
+      [:a.main-icon {:href     "#"
+                     :on-click (fn [_] (set! (.-location js/document) "/"))}
+       [:img {:src "img/icons/nap-logo.svg"}]]
+     [:li {:style {:float "right"}}
+      [ui/icon-button {:on-click #(e! (ts/->ModifyTransportService nil))
+                       :style {:padding 8
+                               :width 60
+                               :height 60}
+                       :icon-style {:height 40
+                                    :width 40}}
+       [ic/action-reorder {:style {:color "#fff"
+                                   :width 40
+                                   :height 40
+                                   }}]]]
+     ]]
+
+  )
+
+
 (defn- top-nav [e! app]
   [:div {:class "topnav"}
-   [:a.main-icon {:href "#"
-                  :on-click #(e! (fp-controller/->ChangePage :front-page))}
-    [:img {:src "img/icons/nap-logo.svg" }]]
-   [:a.ote-nav
-    {:class (is-topnav-active :front-page (:page app))
-     :href "#"
-     :on-click #(e! (fp-controller/->ChangePage :front-page))}
-    (tr [:common-texts :navigation-front-page]) ]
-   [:a.ote-nav
-    {:class (is-topnav-active :own-services (:page app))
-     :href "#"
-     :on-click #(e! (fp-controller/->ChangePage :own-services))}
-    (tr [:common-texts :navigation-own-service-list]) ]
-   [:div.user-menu {:class (is-user-menu-active app) }
-    (r/as-element (user-menu e! (get-in app [:user :name])))]])
+   [:div.container-fluid
+
+      (if (< 900 (get app :width))
```

Comment:
I think it would be easier to read the condition as `(> (:width app) 900)` and also maybe the magic number `900` should be moved to a style ns and given a name (like `mobile-width-px`?)

Reply:
Changed